### ISSUE TITLE
TiDB Crashing on Import

### ISSUE THREAD

**Issue by iliesh** [pingcap/tidb#48615]
### 1. Minimal reproduce step (Required)
I've tried to import a 20GB SQL File

### 2. What did you expect to see? (Required)
Import successfully

### 3. What did you see instead (Required)
```
Connect TiDB:    mysql --comments --host 127.0.0.1 --port 4000 -u root
TiDB Dashboard:  http://127.0.0.1:2379/dashboard
Grafana:        http://127.0.0.1:3000

tidb quit: signal: killed
[2023/11/15 02:26:00.592 +00:00] [ERROR] [tso_dispatcher.go:202] ["[tso] tso request is canceled due to timeout"] [dc-location=global] [error="[PD:client:ErrClientGetTSOTimeout]get TSO timeout"]
[2023/11/15 02:26:12.487 +00:00] [ERROR] [region_cache.go:2602] ["loadStore from PD failed"] [id=1] [error="rpc error: code = DeadlineExceeded desc = context deadline exceeded"]
[2023/11/15 02:26:27.259 +00:00] [INFO] [pd_service_discovery.go:524] ["[pd] cannot update member from this address"] [address=http://127.0.0.1:2379] [error="[PD:client:ErrClientGetMember]error:rpc error: code = DeadlineExceeded desc = context deadline exceeded target:127.0.0.1:2379 status:READY: error:rpc error: code = DeadlineExceeded desc = context deadline exceeded target:127.0.0.1:2379 status:READY"]
[2023/11/15 03:09:38.132 +00:00] [WARN] [region_cache.go:2810] ["[health check] failed to re-resolve unhealthy store"] [error="rpc error: code = DeadlineExceeded desc = context deadline exceeded"]
[2023/11/15 02:27:47.193 +00:00] [ERROR] [region_cache.go:2602] ["loadStore from PD failed"] [id=1] [error="rpc error: code = DeadlineExceeded desc = context deadline exceeded"]
[2023/11/15 03:43:39.514 +00:00] [ERROR] [tso_dispatcher.go:493] ["[tso] getTS error"] [dc-location=global] [stream-addr=http://127.0.0.1:2379] [error="[PD:client:ErrClientGetTSO]get TSO failed, after processing requests"]
[2023/11/15 04:13:45.396 +00:00] [WARN] [util.go:306] ["etcd-cli put kv failed"] [category=ddl] [key=/tidb/server_id/1789] [value=0] [error="context deadline exceeded"] [retryCnt=0]
[2023/11/15 04:12:48.547 +00:00] [ERROR] [pd.go:236] ["updateTS error"] [txnScope=global] [error="rpc error: code = Canceled desc = context canceled"]
[2023/11/15 04:13:23.836 +00:00] [WARN] [coprocessor.go:445] ["buildCopTasks takes too much time"] [elapsed=1h47m5.431291609s] ["range len"=5] ["task len"=1]
[2023/11/15 04:16:54.858 +00:00] [ERROR] [pd_service_discovery.go:257] ["[pd] failed to update member"] [urls="[http://127.0.0.1:2379]"] [error="[PD:client:ErrClientGetMember]get member failed"]
```

### 4. What is your TiDB version? (Required)

v7.4.0
```
~ tiup playground --tag db1
tiup is checking updates for component playground ...
Starting component `playground`: /home/ilie/.tiup/components/playground/v1.13.1/tiup-playground --tag db1
Using the version v7.4.0 for version constraint "".

If you'd like to use a TiDB version other than v7.4.0, cancel and retry with the following arguments:
	Specify version manually:   tiup playground <version>
	Specify version range:      tiup playground ^5
	The nightly version:        tiup playground nightly

Start pd instance:v7.4.0
Start tikv instance:v7.4.0
Start tidb instance:v7.4.0
Waiting for tidb instances ready
127.0.0.1:4000 ... Done
Start tiflash instance:v7.4.0
Waiting for tiflash instances ready
127.0.0.1:3930 ... Done

🎉 TiDB Playground Cluster is started, enjoy!
```

**Comment by lance6716** [comment]
Hi, how do you import the 20GB file, by LOAD DATA or some mysql utility tools?

**Comment by iliesh** [comment]
@lance6716 - importing data into tidb using mysql command line:
```
$ mysql --comments --host 127.0.0.1 --port 4000 -u root
$ mysql> use db test;
$ mysql> source dbdump.sql
```

in the log file I noticed some weird messages (not sure that this is related to the issue tho):
```
[2023/11/20 14:41:34.842 +00:00] [INFO] [scheduler.rs:748] ["get snapshot failed"] [err="Error(Request(message: \"EpochNotMatch current epoch of region 217 is conf_ver: 1 version: 64, but you sent conf_ver: 1 version: 63\" epoch_not_match { current_regions { id: 217 start_key: 7480000000000000FF695F698000000000FF0000070173767230FF382D636FFF726531FF2D31353830FF3831FF393935332E36FF39FF34393400000000FBFF038000000017AC89FFD700000000000000F8 end_key: 7480000000000000FF695F698000000000FF00000D0131303533FF352D4B4CFF2D3430FF6639633039FF392DFF303562336537FF33FF6233407374616EFFFF646172642E69652EFFFF76696174656C2EFF63FF6F6D00000000FF0000F90380000000FF17AC461E00000000FB region_epoch { conf_ver: 1 version: 64 } peers { id: 218 store_id: 1 } } current_regions { id: 219 start_key: 7480000000000000FF6900000000000000F8 end_key: 7480000000000000FF695F698000000000FF0000070173767230FF382D636FFF726531FF2D31353830FF3831FF393935332E36FF39FF34393400000000FBFF038000000017AC89FFD700000000000000F8 region_epoch { conf_ver: 1 version: 64 } peers { id: 220 store_id: 1 } } }))"] [cid=23778]
```
see attached log file before crash:
[crash.log](https://github.com/pingcap/tidb/files/13416488/crash.log)

**Comment by lance6716** [comment]
I see the log you provided is TiKV's log. It does not contain some clue about the cause. `tidb quit: signal: killed` maybe it indicates that the TiDB is killed by OS OOM killer.

**Comment by iliesh** [comment]
While I cannot definitively confirm this at the moment, there seems to be a possibility that the PC slowdown leading to the crash might be attributed to the OM killer potentially terminating the application. This raises the crucial question of why such an event would occur. Our current objective is to transition from the existing MariaDB cluster to TiDB, and it is very important to preemptively address any issues that may arise.

To facilitate this testing, I created a local container with MariaDB and loaded an identical dump file. After a few hours, the file was successfully imported without any apparent issues. Subsequently, I explored another scenario by employing TiDB on a local machine through the tiup playground. During the initial stages of file import, I observed a modest increase in disk activity, but it remained within manageable levels, and the system retained its responsiveness.

However, as time progressed, I encountered a significant slowdown, ultimately leading to the crash of TiKV. This development has raised concerns, especially considering the intention to replace the current MariaDB cluster with TiDB and the desire for a stable and reliable solution. 

To address this, I am seeking insights into the root cause of this performance degradation and crash. Any guidance or recommendations on identifying and resolving these issues would be greatly appreciated.
